designdocs/batched_messages.md

```diff
@@ -10,98 +10,198 @@
-    * `file_size`: `usize`, the size of the batch file, used to determine message fee as well as to let consumers know what size files to expect.  Must be &gt;= the minimum possible DSNP batch file size.
-    * `file_format`: `BatchFormat`, indicator of the file format of the batch file.
-
-The file hash is not included separately. Since the `batch_uri` uses CIDv1 specification, the file hash is already included.
+* `Schema<T:Config>`: generic
+    * `model_type`: `ModelType` See enum section above.
+    * `model`: `Model` Defines the shape of the message payload.
+    * `payload_location`: `PayloadLocation` See enum section above.
 
-* `BatchAnnouncement`: implements `BatchAnnouncementParams`, returned by `get_batches`
+* `Message<T:Config>`: generic
+    * `schema_id`: `u16`
+    * `source`: `MsaId` Source of the message.
+    * `provider`: `MsaId` Public key of a capacity-providing account
+    * `payload`: `Payload` The payload.
 
-See the [implementation of paging in the messages pallet](https://github.com/LibertyDSNP/mrc/blob/main/common/primitives/src/messages.rs#L26-L58) for comparison.
-
-* `BatchAnnouncementOptions<T:Config>`
-    * `msa_id`:  `Option<MsaId>`, the announcer's MSA id.  Pass None() to get all announcements.
-    * `from_block`: `<T::BlockNumber>`, retrieve messages starting at the given block number (inclusive)
-    * `to_block`: `<T::BlockNumber>`, retrieve messages ending at the given block number (inclusive)
-    * `from_index`: `u32`, starting message index
-    * `page_size`: `usize`, retrieve `page_size` messages at a time, up to configured `T::BatchPageSizeMax`. If 0, return `T::BatchPageSizeMax` results
-
-* `BatchAnnouncementResult`
-    * `has_next`: `uint32`, current page number
-    * `next_block`: `<T::BlockNumber>` starting block number of next page of results
-    * `next_index`: `u32` starting index of next results in `next_block`
-    * `results`: `Vec<BatchAnnouncement>`
+(See alternatives section for another way to structure payloads)
 
 ### Extrinsics
-#### announce_batch(origin, batch_announcement_params)
-Creates and posts a new batch announcement message on chain, using the batch message Schema Id. This Schema Id must already be registered and will need to be fetched by the extrinsic.  The transaction fee is determined in part by the file size.
+#### register_schema(origin, schema_params)
+Creates and posts a new schema on chain. The transaction fee is determined in part by the model size.
 
 * **Parameters**
   * origin:  required for all extrinsics, the caller/sender.
-  * `batch_announcement_params`: `BatchAnnouncementParams`, the parameters to use in the batch announcement.
+  * `schema_params`: `Schema`, the parameters to use in the batch announcement.
 
-* **Event**:  `Event::<T>::BatchAnnounced(schema_id, msa_id, file_size, batch_uri)`
 * **Restrictions**:
-  * origin's `msa_id` must have capacity to post the transaction during the current epoch
+  * TBD
 
 ### Custom RPCs
```

Comment:
I think we need to understand how `message.get_messages_by_schema` rpc will change.

Reply:
if I understand correctly, it may not need to change? It looks like the only `Message` related struct is `MessageResponse`, which will keep the same structure it has now. Please let me know if there's something I'm missing.

Reply:
Hmm... I think that is correct for now. We can improve that later.